### ISSUE TITLE
Support for SigmoidCrossEntropyLoss ?

### ISSUE THREAD

**Issue by hassony2** [chuckcho/video-caffe#81]
Hi ! 
First, thanks for the great work.
I'm pretty new to Caffe and I am trying to finetune the pretrained model c3d_ucf101_iter_38000.caffemodel on a binary classification problem with a cross entropy loss.

For this, I modified the last layer to apply SigmoidCrossEntropyLoss, and modified the fc8 layer to have 2 outputs

```
layer {
  name: "my-fc8"
  type: "InnerProduct"
  bottom: "fc7"
  top: "fc8"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output:2
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layer {
  name: "prob"
  type: "Sigmoid"
  bottom: "fc8"
  top: "prob"
  include {
    phase: TEST
  }
}
layer {
  name: "accuracy"
  type: "Accuracy"
  bottom: "prob"
  bottom: "label"
  top: "accuracy/top-1"
  include {
    phase: TEST
  }
}
layer {
  name: "loss"
  type: "SigmoidCrossEntropyLoss"
  bottom: "fc8"
  bottom: "label"
  top: "loss"
}
```

I encounter the following error :  

```
the  Check failed: bottom[0]->count() == bottom[1]->count() (100 vs. 50) SIGMOID_CROSS_ENTROPY_LOSS layer inputs must have the same count.
```

My file containing the path to the training examples looks like this : 

```
/path/to/video1frames 153 0
/path/to/video1frames 157 0
/path/to/video2frames 161 1
/path/to/video2frames 165 1
```

Changing num_output to 1 removes the error, but I gather that in this case it only predicts the score for one class.

I would be gratefull if somoene would share their opinion on what is causing this error and how to fix it.

**Comment by chuckcho** [comment]
@hassony2 In your case, I believe it's multi-class problem where "0"-class and "1"-class can occur at the same time. That's the point of using `SigmoidCrossEntropyLoss` layer (http://caffe.berkeleyvision.org/doxygen/classcaffe_1_1SigmoidCrossEntropyLossLayer.html#details). A label needs to be formed as a binary format "1" for a particular class being present. Note that `video_data_layer` (https://github.com/chuckcho/video-caffe/blob/master/src/caffe/layers/video_data_layer.cpp) assumes one and only class, so it will needs some modification in `video_data_layer`.

**Comment by chuckcho** [comment]
Also this may be of value: http://stackoverflow.com/a/36796252 ;)
